### ISSUE TITLE
IndexError: tuple index out of range with pandas 0.25.

### ISSUE THREAD

**Issue by Foadsf** [matplotlib/matplotlib#14992]
### Bug report

**Bug summary**

I have reported the issue in details [here](https://stackoverflow.com/questions/57372557/matplotlib-drops-indexerror-tuple-index-out-of-range-after-upgrade)
basically after an upgrade to `3.1.1` it drops the error:

> IndexError: tuple index out of range

all other details inclduing the CSV file and code are in the above link.

**Comment by jamespreed** [comment]
Looks like this [commit ](https://github.com/matplotlib/matplotlib/commit/4da5bc1ef7f32607e6028a3d66b5e1eca0910977) changed how the indices were assigned by removing the `_xy_from_xy` method.   This works as of 3.1.0, but not 3.1.1 (on Windows).

**Comment by Foadsf** [comment]
Downgrading to the previous versions doesn't help! There might be more issues IMHO ¯\\\_(ツ)\_/¯

**Comment by anntzer** [comment]
This is actually due to the upgrade to pandas 0.25:
```
import pandas as pd
df = pd.DataFrame({"a": [1, 2, 3]})
print(df.index.shape, df.index[:, None].shape)
```
This now prints `(3,) (3,)`, but with pandas 0.24 used to print `(3,) (3, 1)` which we relied on to convert input to 2D.

**Comment by Foadsf** [comment]
@anntzer downgrading to Pandas `0.24` solved the problem!

**Comment by Foadsf** [comment]
I reported an issue on pandas rep [here](https://github.com/pandas-dev/pandas/issues/27775#issuecomment-518621461)

**Comment by jorisvandenbossche** [comment]
@story645 I am not fully sure we can provide an upstream fix (or at least not long term).

`df.index` is a one-dimensional object, so `df.index[:, None]` actually giving an Index that is 2D, could have been considered a bug in pandas.

**Comment by tacaswell** [comment]
@Foadsf Can you please paste a minimal copy-pastable example into this issue?  SO links rot over time.

**Comment by story645** [comment]
@jorisvandenbossche wondering then if pandas should be throwing a warning or something since 
 `array[:,None]` syntax is the numpy recommended way of expanding dims and so the expectation there is to return a 2D item.

**Comment by jklymak** [comment]
What does np.newaxis give if we use that instead?

**Comment by Foadsf** [comment]
Dear @tacaswell the Pandas guys were kind to edit [my post here](https://github.com/pandas-dev/pandas/issues/27775#issue-477311530) and add clear and concise examples demonstrating the issue.

**Comment by jorisvandenbossche** [comment]
@Foadsf that's a reproducible example for just the pandas side, but I think @tacaswell is looking for an example that shows the error with matplotlib.

Something like

```
s = pd.Series([1, 2, 3])
plt.plot(s.index, s.values)
```

**Comment by tacaswell** [comment]
Thanks @jorisvandenbossche 

@Foadsf in general, you will get much better responses from projects if you provide an easily understandable and testable examples (see http://www.sscce.org/ for a longer discussion as to why).  The information we ask for in the issue template to guide you so we can help you as easily as possible.

**Comment by QuLogic** [comment]
@jklymak `np.newaxis is None`; there is no difference between the two other than programmer clarity.

**Comment by ImportanceOfBeingErnest** [comment]
```
df = pd.DataFrame({"XX" : [4,5,6], "YY" : [7,1,2]})
print(df.index[:,None].shape)
print(df["YY"][:,None].shape)
```

prints 

```
(3,)
(3, 1)
```

This seems rather inconsistent. 

Marking as release critical, because *something* needs to be done here.

**Comment by jorisvandenbossche** [comment]
> This seems rather inconsistent.

@ImportanceOfBeingErnest yes, that is certainly inconsistent, and "kind of" a regression in pandas (which I also propose to fix in pandas itself for 0.25.1, see https://github.com/pandas-dev/pandas/issues/27775#issuecomment-519452205) 
I am saying "kind of" because we should decide what to do when you end up with a 2D Index (which is invalid, but currently you can get that with the indexing operation as `idx[:, None]`): return a 2D array or raise an error. 
`series[:, None]` does return a 2D numpy array, hence you get the correct shape. 

But in any case, I think it would be good for matplotlib to not rely on this indexing aspect to convert a pandas Index into a 2D array-like.
